schoenrock-rossiter_tom_ndvi.py

```diff
@@ -0,0 +1,656 @@
+# 2. Someone else to understand your workflow.
+# 3. The LEAST and most efficient (i.e. runs fast, minimize repetition) amount of code that completes the task.
+# 
+# ### HINTS
+# 
+# * Remove values outside of the landsat valid range of values as specified in the metadata, as needed.
+# * Keep any output files SEPARATE FROM input files. Outputs should be created in an outputs directory that is created in the code (if needed) and/or tested for.
+# * Use the functions that we demonstrated during class to make your workflow more efficient.
+# * BONUS - if you  chose - you can export your data as a csv file. You will get bonus points for doing this.
+# 
+# 
+# ## Assignment Requirements
+# 
+# Your submission to the GitHub repository should include:
+# * This Jupyter Notebook file (.ipynb) with:
+#     * The code to create a plot of mean NDVI across a year for  2 NEON Field Sites:
+#         * NDVI on the x axis and formatted dates on the y for both NEON sites on one figure/axis object
+#     * The **data should be cleaned to remove the influence of clouds**. See the [earthdatascience website for an example of what your plot might look like with and without removal of clouds](https://www.earthdatascience.org/courses/earth-analytics-python/create-efficient-data-workflows/).
+# * BONUS: Create one output `.csv` file that has 3 columns - NDVI, Date and Site Name - with values for SJER and HARV.
+# 
+# Your notebook should:
+# * Have *at least* 2 well documented and well named functions with docstrings.
+# * Include a Markdown cell at the top of the notebook that outlines the overall workflow using pseudocode (i.e. plain language, not code)
+# * Include additional Markdown cells throughout the notebook to describe: 
+#     * the data that you used - and where it is from
+#     * how data are being processing
+#     * how the code is optimized to run fast and be more concise
+
+# # Replace this cell with your pseudocode  for this workflow
+# 
+# If you happen to be a diagram person a diagram is ok too
+# 
+# 
+
+# Workflow pseudocode
+# 
+# Run first for one month at one site - next run for all months in one site - last run for all months in both sites
+# 
+# Preparation - 
+# - Import packages
+# - Set working directory to 'data'
+# - Download data from earthpy
+# 
+# Access Landsat band files for single scene -
+# - Create path to Landsat data
+# - Create list of Landsat bands in scene (into df?)
+# - Subset bands into just RGB, NIR
+# - Stack bands into single raster
+# - clip stacked raster by boundary layer
+# - Output raster for individual month
+# 
+# Access Landsat band files for all scenes in a site -
+# - Get list of all directories
+# - For bands in each scene directory perform the above steps
+# - Output, into a list or df, the average ndvi and data for each scene
+# 
+# Access Landsat band files for all scenes in both sites -
+# 
+# Calculate ndvi -
+# - Run ndvi expression on red and NIR bands
+# - Calculate mean ndvi for site
```

Comment:
Wouldn't it be more accurate to say "calculate mean ndvi for each scene for both sites"?

Reply:
Another good spot, indeed that would enhance clarity.